addons/udes_stock/models/stock_picking.py

```diff
@@ -1707,15 +1711,22 @@ def reserve_stock(self):
             # should not reserve stock if the batch cannot be completely
             # reserved.
             to_reserve = picking_type.u_num_reservable_pickings
-            reserve_all = picking_type.u_num_reservable_pickings == -1
+            reserve_all = to_reserve == -1
             base_domain = [
                 ('picking_type_id', '=', picking_type.id),
                 ('state', '=', 'confirmed')]
             limit = 1
             processed = Picking.browse()
             by_type = lambda x: x.picking_type_id == picking_type
 
-            while to_reserve > 0 or reserve_all:
+            if not reserve_all:
+                ready_pickings = Picking.search([
+                    ('picking_type_id', '=', picking_type.id),
+                    ('state', '=', 'assigned'),
+                ])
+                to_reserve -= len(ready_pickings)
```

Comment:
Could we use [search_count](https://github.com/odoo/odoo/blob/11.0/odoo/models.py#L1451), as we aren't using `ready_pickings` for anything else?

Reply:
Yes, we do.